pkg/backend/display/diff.go

```diff
@@ -256,8 +256,20 @@ func renderDiffResourcePreEvent(
 	if shouldShow(payload.Metadata, opts) || isRootStack(payload.Metadata) {
 		indent := engine.GetIndent(payload.Metadata, seen)
 		summary := engine.GetResourcePropertiesSummary(payload.Metadata, indent)
-		details := engine.GetResourcePropertiesDetails(
-			payload.Metadata, indent, payload.Planning, opts.SummaryDiff, payload.Debug)
+
+		var details string
+		if payload.Metadata.RichDiff != nil {
+			var buf bytes.Buffer
+			if diff := translateRichDiff(payload.Metadata); diff != nil {
+				engine.PrintObjectDiff(&buf, *diff, nil, payload.Planning, indent, opts.SummaryDiff, payload.Debug)
+			} else {
+				engine.PrintObject(&buf, payload.Metadata.Old.Inputs, payload.Planning, indent, deploy.OpSame, true, payload.Debug)
```

Comment:
consider named arg for `true`.  it's not clear at the callsite what that indicates.